parlai/crowdsourcing/tasks/turn_annotations_static/webapp/src/components/task_components.jsx

```diff
@@ -184,125 +49,352 @@ function SubtaskCounter({ subtaskIndex, numSubtasks }) {
-                annotationBuckets={annotationBuckets}
-                doAnnotateMessage={m.do_annotate}
-                askReason={askReason}
-                responseField={responseField}
-                onUserInputUpdate={() => handleUserInputUpdate(subtaskData)}
-              />
-            </div>
+      {subtaskData.map((m, idx) => (
+        <div key={idx}>
+          <div>
+            <ChatMessage
+              subtaskIdx={subtaskIndex}
+              turnIdx={idx}
+              turnAnswers={m}
+              annotationQuestion={annotationQuestion}
+              annotationBuckets={annotationBuckets}
+              doAnnotateMessage={m.do_annotate}
+              askReason={askReason}
+              askResponse={taskConfig.response_field}
+              setChecked={setChecked}
+              setReason={setReason}
+              setResponse={setResponse}
+            />
           </div>
-      )}
+        </div>
+      ))}
     </div>
-  )
+  );
 }
 
-function MainTaskComponent({ taskData, taskTitle, taskDescription, taskConfig, onSubmit }) {
-  if (taskData == undefined) {
-    return <div><p> Loading chats...</p></div>;
+// type Buckets = {
+//   [bucket: string]: ?boolean,
+// };
+
+// type TurnAnswers = {
+//   text: string,
+//   agent_idx: number,
+//   other_metadata: any,
+//   buckets: Buckets,
+//   do_annotate: boolean,
+//   input_reason: ?string,
+//   response: ?string,
+// };
+
+// type SubtaskAnswers = Array<TurnAnswers>;
+// type WorkerAnswers = Array<SubtaskAnswers>;
+
+// type MainTaskComponentProps = {
+//   taskData: TaskData,
+//   taskTitle: string,
+//   taskDescription: string,
+//   taskConfig: TaskConfig,
+//   onSubmit: WorkerAnswers => void,
+// };
+// type MainTaskComponentState = {
+//   index: number,
+//   answers: WorkerAnswers,
+// };
+
+class MainTaskComponent extends React.Component {
+  state = {
+    index: 0,
+    answers: this._initAnswersFromTaskData(),
+  };
+  constructor(props) {
+    super(props);
   }
-  const [index, setIndex] = React.useState(0);
 
-  return (
-    <div style={{ margin: 0, padding: 0, height: '100%' }}>
-      <LeftPane>
-        <h4><span dangerouslySetInnerHTML={{ __html: taskTitle || 'Task Title Loading' }}></span></h4>
-        <SubtaskCounter subtaskIndex={index} numSubtasks={taskData.length}></SubtaskCounter>
+  _initAnswersFromTaskData() {
+    var answers = [];
+    for (var i = 0; i < this.props.taskData.length; i++) {
+      var answersForSubtaskIndex = [];
+      const subTaskData = this.props.taskData[i];
+      for (var j = 0; j < subTaskData.length; j++) {
+        var answersForTurn = {
+          text: subTaskData[j].text,
+          agent_idx: subTaskData[j].agent_idx,
+          other_metadata: subTaskData[j].other_metadata,
+          buckets: {},
+          do_annotate: subTaskData[j].do_annotate,
+          input_reason: null,
+          response: null,
+        };
+        Object.keys(this.props.taskConfig.annotation_buckets.config).forEach(
```

Comment:
Null check around this since annotation_buckets field is optional so it can be null.